lib/rouge/lexers/powershell.rb

```diff
@@ -13,21 +13,49 @@ class Powershell < Shell
       filenames '*.ps1', '*.psm1', '*.psd1', '*.psrc', '*.pssc'
       mimetypes 'text/x-powershell'
 
+      # https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_commonparameters?view=powershell-6
       ATTRIBUTES = %w(
         CmdletBinding ConfirmImpact DefaultParameterSetName HelpURI SupportsPaging
         SupportsShouldProcess PositionalBinding
       ).join('|')
-
+      
+      # https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_reserved_words?view=powershell-6
       KEYWORDS = %w(
-        Begin Exit Process Break Filter Return Catch Finally Sequence Class For
-        Switch Continue ForEach Throw Data From Trap Define Function Try Do If
-        Until DynamicParam In Using Else InlineScript Var ElseIf Parallel While
-        End Param Workflow
+        assembly exit process
+        base filter public
+        begin finally return
+        break for sequence
+        catch foreach static
+        class from switch
+        command function throw
+        configuration hidden trap
+        continue if try
+        data in type
+        define inlinescript until
+        do interface using
+        dynamicparam module var
+        else namespace while
+        elseif parallel workflow
+        end param
+        enum private
       ).join('|')
-
+      
+      # https://devblogs.microsoft.com/scripting/powertip-find-a-list-of-powershell-type-accelerators/
+      # ([PSObject].Assembly.GetType("System.Management.Automation.TypeAccelerators")::Get).Keys -join ' '
       KEYWORDS_TYPE = %w(
```

Comment:
these are  what ms calls type accelerators, as mentioned in the link. They are keywords that tell powershell what type of object a thing is. They are, in fact, types. they're just more complex types that you might be used to seeing in other languages. python for example has a smaller set of basic types. Probably because underneath the hood powershell is a lot like C#.

Reply:
I guess more specifically what I was questioning were the accelerators `AllowEmptyCollection`, `AllowEmptyString`, `AllowNull`, `ArgumentComplete`, `ValidateCount`, `ValidateDrive`, `ValidateLength`, `ValidateNotNull`, `ValidateNotNullOrEmpty`, `ValidatePattern`, `ValidateRange`, `ValidateScript`, `ValidateSet`, `ValidateTrustedData` and `ValidateUserDrive`. 

I'm struggling to understand what an `AllowEmptyCollection` type is, for example.

Reply:
https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_functions_advanced_parameters?view=powershell-6

They are validation types.  I don't see any other relevant category for them in the legend.

Reply:
Ah, I get it. Sorry it took a while >_<

Reply:
Not a problem. A part of your job is to make sure the lexer is good, and so you're simply doing your job.